app/stats/[slug]/csv/route.ts

```diff
@@ -0,0 +1,39 @@
+import { usePapaParse } from "react-papaparse";
+
+import { getAllMetricSamples, type MetricSample } from "~/src/data/metrics";
+import { ContentType } from "~/src/utils";
+
+type Props = {
+  params: {
+    slug: string;
+  };
+};
+
+export async function GET(_: Request, { params }: Props): Promise<Response> {
+  const { slug } = params;
+
+  const { jsonToCSV } = usePapaParse();
+
+  const compareSamplesByDate = (a: MetricSample, b: MetricSample) =>
+    new Date(a.date).getTime() - new Date(b.date).getTime();
+
+  const allSamples = await getAllMetricSamples();
+  const samples = allSamples
```

Comment:
Tohle opakujeme hodně často – v titulní stránce metrik, v detailu metriky a teď i tady. V tomhle PR bych to neřešil, ale výhledově bychom měli přidat třeba do `src/data/Metrics.ts` pár helperů na tyhle věci, například načtení všech vzorků pro danou metriku.